airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -716,21 +716,37 @@ def get_updated_state(
         """
         updated_state = super().get_updated_state(current_stream_state, latest_record)
         if self.parent_stream_class:
+            parent_state = latest_record.get(self.parent_stream.name, {})
+            parent_state_value = (
+                parent_state.get(self.parent_stream.cursor_field) if parent_state else latest_record.get(self.parent_stream.cursor_field)
+            )
             # add parent_stream_state to `updated_state`
-            updated_state[self.parent_stream.name] = {self.parent_stream.cursor_field: latest_record.get(self.parent_stream.cursor_field)}
+            updated_state[self.parent_stream.name] = {self.parent_stream.cursor_field: parent_state_value}
         return updated_state
 
-    def get_stream_state_value(self, stream_state: Optional[Mapping[str, Any]]) -> str:
-        if self.parent_stream_class:
-            # get parent stream state from the stream_state object.
-            parent_state = stream_state.get(self.parent_stream.name, {})
-            if parent_state:
-                return parent_state.get(self.parent_stream.cursor_field, self.default_state_comparison_value)
-        else:
-            # get the stream state, if no `parent_stream_class` was assigned.
+    def _get_stream_cursor_value(self, stream_state: Optional[Mapping[str, Any]] = None) -> Optional[str]:
+        if stream_state:
             return stream_state.get(self.cursor_field, self.default_state_comparison_value)
+        else:
+            return self.config.get("start_date")
+
+    def get_stream_state_value(self, stream_state: Optional[Mapping[str, Any]] = None) -> Optional[str]:
+        if stream_state:
+            if self.parent_stream_class:
+                # get parent stream state from the stream_state object.
+                parent_state = stream_state.get(self.parent_stream.name, {})
+                if parent_state:
+                    return parent_state.get(self.parent_stream.cursor_field, self.default_state_comparison_value)
+                else:
+                    # use the streams cursor value, if no parent state available
+                    return self._get_stream_cursor_value(stream_state)
```

Comment:
Can we document the impact of reverting if necessary. My understanding is that things should be fine since we call `updated_state = super().get_updated_state(current_stream_state, latest_record)` and the state will now have two part: the parent and the child. If we revert, the child is still populated so we should be fine

Reply:
I'll update the PR description, once get back to this PR later today, thanks @maxi297